keel-slack/src/main/kotlin/com/netflix/spinnaker/keel/slack/handlers/LifecycleEventNotificationHandler.kt

```diff
@@ -0,0 +1,62 @@
+package com.netflix.spinnaker.keel.slack.handlers
+
+import com.netflix.spinnaker.keel.api.ScmInfo
+import com.netflix.spinnaker.keel.lifecycle.LifecycleEventType
+import com.netflix.spinnaker.keel.notifications.NotificationType
+import com.netflix.spinnaker.keel.slack.SlackLifecycleNotification
+import com.netflix.spinnaker.keel.slack.SlackService
+import com.slack.api.model.kotlin_extension.block.withBlocks
+import org.apache.logging.log4j.util.Strings
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.stereotype.Component
+
+/**
+ * Sends notification for a lifecycle event, like bake / build failures
+ */
+@Component
+class LifecycleEventNotificationHandler (
+  private val slackService: SlackService,
+  @Value("\${spinnaker.baseUrl}") private val spinnakerBaseUrl: String,
+  private val gitDataGenerator: GitDataGenerator
+) : SlackNotificationHandler<SlackLifecycleNotification> {
+
+  override val type: NotificationType = NotificationType.LIFECYCLE_EVENT
+  private val log by lazy { LoggerFactory.getLogger(javaClass) }
+
+  override fun sendMessage(notification: SlackLifecycleNotification) {
+    log.debug("Sending pinned artifact notification for application ${notification.application}")
+
+    with(notification) {
+      val imageUrl = when (type) {
+          LifecycleEventType.BAKE -> "https://raw.githubusercontent.com/gcomstock/managed.delivery/master/src/icons/bake_fail.png"
```

Comment:
We could host these in `spinnaker.io` if you don't want to use Greg's personal GitHub URL.

Reply:
I just spoke with @erikmunson about it. We agreed it can be hosted on GitHub for now (as it's not on VPN, and does not require auth), and we can refactor it later as we think of where can be the best place that we could host those images. I'll mark at as a todo.